modular_skyrat/modules/reagent_forging/code/anvil.dm

```diff
@@ -32,46 +32,62 @@
+	if(!obj_anvil_search && obj_tong_search)
+		obj_tong_search.forceMove(src)
 		update_appearance()
 		forge_item.icon_state = "tong_empty"
 		return FALSE
 
 /obj/structure/reagent_anvil/hammer_act(mob/living/user, obj/item/tool)
+	//regardless, we will make a sound
 	playsound(src, 'modular_skyrat/modules/reagent_forging/sound/forge.ogg', 50, TRUE, ignore_walls = FALSE)
-	var/obj/item/forging/incomplete/search_incomplete_src = locate(/obj/item/forging/incomplete) in contents
-	if(!search_incomplete_src)
-		return FALSE
-	if(COOLDOWN_FINISHED(search_incomplete_src, heating_remainder))
-		to_chat(user, span_warning("You mess up, the metal was too cool!"))
+	//do we have an incomplete item to hammer out? if so, here is our block of code
+	var/obj/item/forging/incomplete/search_incomplete_src = locate() in contents
+	if(search_incomplete_src)
+		if(COOLDOWN_FINISHED(search_incomplete_src, heating_remainder))
+			to_chat(user, span_warning("You mess up, the metal was too cool!"))
+			search_incomplete_src.times_hit -= 3
+			return FALSE
+		if(COOLDOWN_FINISHED(search_incomplete_src, striking_cooldown))
+			var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * search_incomplete_src.average_wait
+			COOLDOWN_START(search_incomplete_src, striking_cooldown, skill_modifier)
+			search_incomplete_src.times_hit++
+			balloon_alert(user, "good hit!")
+			user.mind.adjust_experience(/datum/skill/smithing, 1) //A good hit gives minimal experience
+			if(search_incomplete_src?.times_hit >= search_incomplete_src.average_hits)
+				to_chat(user, span_notice("[search_incomplete_src] is sounding ready."))
+			return FALSE
 		search_incomplete_src.times_hit -= 3
+		balloon_alert(user, "bad hit!")
+		if(search_incomplete_src?.times_hit <= -(search_incomplete_src.average_hits))
+			to_chat(user, span_warning("The hits were too inconsistent-- [search_incomplete_src] breaks!"))
+			qdel(search_incomplete_src)
+			update_appearance()
 		return FALSE
-	if(COOLDOWN_FINISHED(search_incomplete_src, striking_cooldown))
-		var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER) * search_incomplete_src.average_wait
-		COOLDOWN_START(search_incomplete_src, striking_cooldown, skill_modifier)
-		search_incomplete_src.times_hit++
-		balloon_alert(user, "good hit!")
-		user.mind.adjust_experience(/datum/skill/smithing, 1) //A good hit gives minimal experience
-		if(search_incomplete_src?.times_hit >= search_incomplete_src.average_hits)
-			to_chat(user, span_notice("[search_incomplete_src] is sounding ready."))
-		return FALSE
-	search_incomplete_src.times_hit -= 3
-	balloon_alert(user, "bad hit!")
-	if(search_incomplete_src?.times_hit <= -(search_incomplete_src.average_hits))
-		to_chat(user, span_warning("The hits were too inconsistent-- [search_incomplete_src] breaks!"))
-		qdel(search_incomplete_src)
-		update_appearance()
+
+	var/obj/obj_anvil_search = locate() in contents
```

Comment:
Same here, this doesn't really tell me much about the purpose of what we find in that var.